Docs/grafana helm (#80390)
grafana/grafana: docs/sources/setup-grafana/installation/helm/index.md

   ```bash
   helm list -n monitoring
   ```

   You should see an output similar to the following:

   ```bash
   NAME            NAMESPACE       REVISION        UPDATED                                 STATUS          CHART          APP VERSION
   my-grafana      monitoring      1               2024-01-13 23:06:42.737989554 +0000 UTC deployed        grafana-6.59.0 10.1.0   
   ```

5. To check the overall status of all the objects in the namespace, run the following command:

      ```bash
      kubectl get all -n monitoring
      ```

   If you encounter errors or warnings in the **STATUS** column, check the logs and refer to Troubleshooting setction of this documentation.

### Accessing Grafana

This section describes the steps you must complete to access Grafana via web browser.

First, run the following `helm status` command:

```bash
helm status my-grafana -n monitoring
```

This command will give you the complete status of the release information such as last deployment time, namespace (where it is deployed), release status, additional notes provided by the chart etc.

> Within the additonal release notes, it provides the following 2 key information i.e.

Reviewer: @usmangt I'm having trouble following this. When I run the helm status command you say it provides the following 2 key information. Does that command literally have the words "How do decode the login password for the admin account"? Or is there something in that output that the user needs to make note of so that they can use it in the next command, which is the command on like 159?

1. How to decode the login password for the admin account
2. Expose service to the web browser

#### Decoding the admin account password

Run the command as follows (it should be the same command you saw when you executed the `helm status` command):

> ```bash

Reviewer: This command looks very different from the helm status command on line 144-146.

kubectl get secret --namespace monitoring my-grafana -o jsonpath="{.data.admin-password}" | base64 --decode ; echo
```

> It will give you a decoded `base64` string output which is the password for the admin account (note it down by saving it into a file).

Reviewer: The note it down part should be its own step.

#### Access Grafana via the Kubernetes service in a local web browser

1. Follow the instructions as described in the above `helm status` command, which provides complete instructions on accessing the Grafana server on port `3000`.

   ```bash
   # first run the below command export a shell variable named POD_NAME that will save the complete name of the pod which got deployed

   export POD_NAME=$(kubectl get pods --namespace monitoring -l "app.kubernetes.io/name=grafana,app.kubernetes.io/instance=my-grafana" -o jsonpath="{.items[0].metadata.name}")

   # then run the port forwarding command to direct the grafana pod to listen to port 3000

   kubectl --namespace monitoring port-forward $POD_NAME 3000
   ```

   For more information about port-forwarding, refer to [Use Port Forwarding to Access Applications in a Cluster](https://kubernetes.io/docs/tasks/access-application-cluster/port-forward-access-application-cluster/).

2. Navigate to `127.0.0.1:3000` in your browser.
3. The Grafana sign-in page appears.
4. To sign in, enter `admin` for the username.
5. For the password paste it which you have saved to a file after decoding it earlier.

> ## Customizing Grafana default configuraiton

Reviewer: ```suggestion
## Customize Grafana default configuraiton
```

> Helm is a popular package manager for Kubernetes. It bundles Kubernetes resource manifests in a way that they may be re-used across different environments. These manifests are written in a templating language, allowing us to provide configuration values via `values.yaml` file, or in-line using helm, to replace the placeholders in the manifest where these configurations should reside.

Reviewer: ```suggestion
Helm is a popular package manager for Kubernetes. It bundles Kubernetes resource manifests to be re-used across different environments. These manifests are written in a templating language, allowing you to provide configuration values via `values.yaml` file, or in-line using Helm, to replace the placeholders in the manifest where these configurations should reside.
```

By default, persistent storage is disabled, which means that Grafana uses ephemeral storage, and all data will be stored within the container's file system. This data will be lost if the container is stopped, restarted, or if the container crashes.

It is highly recommended that you enable persistent storage in Grafana Helm charts if you want to ensure that your data persists and is not lost in case of container restarts or failures.

Enabling persistent storage in Grafana Helm charts ensures a reliable solution for running Grafana in production environments.

To enable the persistent storage in the Grafana Helm charts, complete the following steps:

1. Open the `values.yaml` file in your favorite editor.

2. Edit the values and under the section of `persistence`, change the `enable` flag from `false` to `true`

   ```yaml
   .......
   ............
   ......
   persistence:
   type: pvc
   enabled: true
   # storageClassName: default
   .......
   ............
   ......
   ```

3. Run the following `helm upgrade` command by specifying the `values.yaml` file to make the changes take effect:

   ```bash
   helm upgrade my-grafana grafana/grafana -f values.yaml -n monitoring
   ```

> After that, the PVC will be enabled and able to store all of your data e.g. Dashboards, Data sources, etc.

Reviewer: ```suggestion
The PVC will now store all your data such as dashboards, data sources, and so on.
```

### Install plugins (e.g. Zabbix app, Clock panel, etc.)

You can install plugins in Grafana from the official and community [plugins page](https://grafana.com/grafana/plugins). These plugins allow you to add new visualization types, data sources, and applications to help you better visualize your data.

Grafana currently supports three types of plugins: panel, data source, and app. For more information on managing plugins, refer to [Plugin Management](https://grafana.com/docs/grafana/latest/administration/plugin-management/).

To install plugins in the Grafana Helm Charts, complete the following steps:

1. Open the `values.yaml` file in your favorite editor.

> 2. Find the line that says `plugins:` and under that section, define the plugins that you would like to install e.g.

Reviewer: ```suggestion
2. Find the line that says `plugins:` and under that section, define the plugins that you want to install.
```

   ```yaml
   .......
   ............
   ......
   plugins:
   # here we are installing two plugins, make sure to keep the indentation correct as written here.

   - alexanderzobnin-zabbix-app
   - grafana-clock-panel
   .......
   ............
   ......
   ```

> 3. After that save the changes and use the `helm upgrade` command to get these plugins installed:

Reviewer: You do not need to ever use the words "After that" in a task because the numbered step implies the concept of "after".

   ```bash
   helm upgrade my-grafana grafana/grafana -f values.yaml -n monitoring
   ```

>    You can verify if the above plugins got installed by login into the Grafana UI -> Administration -> Plugins.

Reviewer: Please format as steps and the admin -> plugins should be admin > Plugins. That how we format navigation.

---
aliases:
  - ../../installation/helm/
description: Guide for deploying Grafana using Helm Charts
labels:
  products:
    - enterprise
    - oss
menuTitle: Grafana on Helm Charts
title: Deploy Grafana using Helm Charts
weight: 500
---

# Deploy Grafana using Helm Charts

> On this page, you will find instructions for installing and running Grafana on Kubernetes using Helm Charts.

Reviewer: ```suggestion
This topic includes instructions for installing and running Grafana on Kubernetes using Helm Charts.
```

> ## Introduction

Reviewer: ```suggestion
```

[Helm](https://helm.sh/) is an open-source command line tool used for managing Kubernetes applications. It is a graduate project in the [CNCF Landscape](https://www.cncf.io/projects/helm/).

{{% admonition type="note" %}}
The Grafana open-source community offers Helm Charts for running it on Kubernetes. Please be aware that the code is provided without any warranties. If you encounter any problems, you can report them to the [Official GitHub repository](https://github.com/grafana/helm-charts/).
{{% /admonition %}}

> ### Before you begin

Reviewer: ```suggestion
## Before you begin
```

{{% admonition type="note" %}}
The Grafana open-source community offers Helm Charts for running it on Kubernetes. Please be aware that the code is provided without any warranties. If you encounter any problems, you can report them to the [Official GitHub repository](https://github.com/grafana/helm-charts/).
{{% /admonition %}}

## Before you begin

To install Grafana using Helm, ensure you have completed the following:

- Install a Kubernetes server on your machine. For information about installing Kubernetes, refer to [Deploy Grafana on Kubernetes](/docs/grafana/latest/setup-grafana/installation/kubernetes/)
- Install the latest stable version of Helm. For information on installing Helm, refer to [Install Helm](https://helm.sh/docs/intro/install/).

## Install Grafana using Helm

When you install Grafana using Helm, you complete the following tasks:

1. Set up the Grafana Helm repository, which provides a space in which you will install Grafana.

1. Deploy Grafana using Helm, which installs Grafana into a namespace.

> 1. Accessing Grafana, which provides steps to sign into Grafana.

Reviewer: ```suggestion
1. Access Grafana, which provides steps to sign into Grafana.
```

   my-grafana      monitoring      1               2024-01-13 23:06:42.737989554 +0000 UTC deployed        grafana-6.59.0 10.1.0   
   ```

1. To check the overall status of all the objects in the namespace, run the following command:

      ```bash
      kubectl get all -n monitoring
      ```

   If you encounter errors or warnings in the **STATUS** column, check the logs and refer to the Troubleshooting section of this documentation.

### Accessing Grafana

This section describes the steps you must complete to access Grafana via web browser.

1. Run the following `helm get notes` command:

   ```bash
   helm get notes my-grafana -n monitoring
   ```

   This command will print out the chart notes. You will the output `NOTES` that provide the complete instructions about:
   - How to decode the login password for the Grafana admin account
   - Access Grafana service to the web browser

1. To get the Grafana admin password, run the command as follows:

   ```bash
   kubectl get secret --namespace monitoring my-grafana -o jsonpath="{.data.admin-password}" | base64 --decode ; echo
   ```

>    It will give you a decoded `base64` string output which is the password for the admin account.

Reviewer: ```suggestion
   This command provides a decoded `base64` string output which is the password for the admin account.
```

1. Save the decoded password to a file on your machine.

> 1. To access Grafana service on the web browser, run the following command:

Reviewer: ```suggestion
1. To access the Grafana service on the web browser, run the following command:
```

   ```bash
   export POD_NAME=$(kubectl get pods --namespace monitoring -l "app.kubernetes.io/name=grafana,app.kubernetes.io/instance=my-grafana" -o jsonpath="{.items[0].metadata.name}")
   ```

>    The above command will export a shell variable named `POD_NAME` that will save the complete name of the pod which got deployed.

Reviewer: ```suggestion
   This command exports a shell variable named `POD_NAME` that saves the name of the pod that got deployed.
```

1. Run the following port forwarding command to direct the Grafana pod to listen to port `3000`:

   ```bash
   kubectl --namespace monitoring port-forward $POD_NAME 3000
   ```

   For more information about port-forwarding, refer to [Use Port Forwarding to Access Applications in a Cluster](https://kubernetes.io/docs/tasks/access-application-cluster/port-forward-access-application-cluster/).

1. Navigate to `127.0.0.1:3000` in your browser.

1. The Grafana sign-in page appears.

1. To sign in, enter `admin` for the username.

> 1. For the password paste it which you have saved to a file after decoding it earlier.

Reviewer: ```suggestion
1. Copy the password from the saved file you previously decoded, and paste it into the **Password** field.
```

## Customize Grafana default configuraiton

> Helm is a popular package manager for Kubernetes. It bundles Kubernetes resource manifests to be re-used across different environments. These manifests are written in a templating language, allowing you to provide configuration values via `values.yaml` file, or in-line using Helm, to replace the placeholders in the manifest where these configurations should reside.

Reviewer: ```suggestion
Helm is a popular package manager for Kubernetes. It bundles Kubernetes resource manifests that you can reuse across different environments. These manifests are written in a templating language, allowing you to provide configuration values via the `values.yaml` file or in-line using Helm to replace the placeholders in the manifest where these configurations should reside.
```

> The `values.yaml` file allows you to customize the chart's configuration by specifying values for various parameters such as image versions, resource limits, service configurations, etc.

Reviewer: ```suggestion
The `values.yaml` file allows you to customize the chart's configuration by specifying values for various parameters such as image versions, resource limits, service configurations, and so on.
```

By modifying the values in the `values.yaml` file, you can tailor the deployment of a Helm chart to your specific requirements by using the helm install or upgrade commands. For more information about configuring Helm, refer to [Values Files](https://helm.sh/docs/chart_template_guide/values_files/).

### Download the values.yaml file

In order to make any configuration changes, download the `values.yaml` file from the Grafana Helm Charts repository:

https://github.com/grafana/helm-charts/edit/main/charts/grafana/values.yaml

{{% admonition type="note" %}}
Depending on your use case requirements, you can use a single YAML file that contains your configuration changes or you can create multiple YAML files.
{{% /admonition %}}

### Enable persistent storage **(recommended)**

> By default, persistent storage is disabled, which means that Grafana uses ephemeral storage, and all data will be stored within the container's file system. This data will be lost if the container is stopped, restarted, or if the container crashes.

Reviewer: ```suggestion
By default, persistent storage is disabled, which means that Grafana uses ephemeral storage, and all data is stored in the container's file system. This data is lost if the container is stopped, restarted, or if the container crashes.
```

> It is highly recommended that you enable persistent storage in Grafana Helm charts if you want to ensure that your data persists and is not lost in case of container restarts or failures.

Reviewer: ```suggestion
It is recommended that you enable persistent storage in Grafana Helm charts if you want to ensure that your data persists and is not lost in case of container restarts or failures.
```

Enabling persistent storage in Grafana Helm charts ensures a reliable solution for running Grafana in production environments.

To enable the persistent storage in the Grafana Helm charts, complete the following steps:

1. Open the `values.yaml` file in your favorite editor.

1. Edit the values and under the section of `persistence`, change the `enable` flag from `false` to `true`

   ```yaml
   .......
   ............
   ......
   persistence:
   type: pvc
   enabled: true
   # storageClassName: default
   .......
   ............
   ......
   ```

> 1. Run the following `helm upgrade` command by specifying the `values.yaml` file to make the changes take effect:

Reviewer: ```suggestion
1. For the changes to take effect, run the following `helm upgrade` command that includes the `values.yaml` file.
```

   ```bash
   helm upgrade my-grafana grafana/grafana -f values.yaml -n monitoring
   ```

The PVC will now store all your data such as dashboards, data sources, and so on.

### Install plugins (e.g. Zabbix app, Clock panel, etc.)

You can install plugins in Grafana from the official and community [plugins page](https://grafana.com/grafana/plugins). These plugins allow you to add new visualization types, data sources, and applications to help you better visualize your data.

Grafana currently supports three types of plugins: panel, data source, and app. For more information on managing plugins, refer to [Plugin Management](https://grafana.com/docs/grafana/latest/administration/plugin-management/).

To install plugins in the Grafana Helm Charts, complete the following steps:

> 1. Open the `values.yaml` file in your favorite editor.

Reviewer: ```suggestion
1. Open the `values.yaml` file in an editor.
```

1. Find the line that says `plugins:` and under that section, define the plugins that you want to install.

   ```yaml
   .......
   ............
   ......
   plugins:
   # here we are installing two plugins, make sure to keep the indentation correct as written here.

   - alexanderzobnin-zabbix-app
   - grafana-clock-panel
   .......
   ............
   ......
   ```

> 1. Save the changes and use the `helm upgrade` command to get these plugins installed:

Reviewer: ```suggestion
1. Save the changes and use the `helm upgrade` command to install the plugins:
```

   ```bash
   helm upgrade my-grafana grafana/grafana -f values.yaml -n monitoring
   ```

1. Navigate to `127.0.0.1:3000` in your browser.

1. Login with admin credentials when the Grafana sign-in page appears.

> 1. Navigate to UI -> Administration -> Plugins

Reviewer: ```suggestion
1. Navigate to **Administration > Plugins**.
```

> 1. Search for the above plugins and they should be marked as installed.

Reviewer: ```suggestion
The plugins you specified should be marked as installed.
```

### Collecting logs

It is important to view the Grafana server logs while troubleshooting any issues.

To check the Grafana logs, run the following command:

```bash
# dump Pod logs for a Deployment (single-container case)

kubectl logs --namespace=monitoring deploy/my-grafana
```

If you have multiple containers running in the deployment, run the following command to obtain the logs only for the Grafana deployment:

```bash
# dump Pod logs for a Deployment (multi-container case)

kubectl logs --namespace=monitoring deploy/grafana -c my-grafana
```

For more information about accessing Kubernetes application logs, refer to [Pods](https://kubernetes.io/docs/reference/kubectl/cheatsheet/#interacting-with-running-pods) and [Deployments](https://kubernetes.io/docs/reference/kubectl/cheatsheet/#interacting-with-deployments-and-services).

### Increasing log levels

By default, the Grafana log level is set to `info`, but you can increase it to `debug` mode to fetch information needed to diagnose and troubleshoot a problem. For more information about Grafana log levels, refer to [Configuring logs](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana#log).

> To increase log level to `debug` mode, use the following steps:

Reviewer: ```suggestion
To increase log level to `debug` mode, complete the following steps:
```

> 1. Open the `values.yaml` file in your favorite editor and search for the string `grafana.ini` and there you will find a section about log mode.

Reviewer: ```suggestion
1. Open the `values.yaml` file in an editor, search for the string `grafana.ini`, and locate the log mode section.
```

1. Add level: `debug` just below the line `mode: console`

   ```yaml
   # This is the values.yaml file
      .....
   .......
   ....
   grafana.ini:
   paths:
      data: /var/lib/grafana/
      .....
   .......
   ....
      mode: console
      level: debug
   ```

>    Make sure to keep the indentation level the same otherwise it will not work.

Reviewer: ```suggestion
   Make sure to keep the indentation level the same; otherwise, it will not work.
```

> 1. Now to apply this, run the `helm upgrade` command as follows:

Reviewer: ```suggestion
1. To apply this change, run the following `helm upgrade` command:
```

   ```bash
   helm upgrade my-grafana grafana/grafana -f values.yaml -n monitoring
   ```

> 1. To verify it, access the Grafana UI in the browser using the provided `IP:Port`. The Grafana sign-in page appears.

Reviewer: ```suggestion
1. To verify the change has taken effect, use the provided `IP:Port` to access the Grafana UI in the browser. 
   
   The Grafana sign-in page appears.
```

> 1. To sign in to Grafana, enter `admin` for the username and paste the password which was decoded earlier. Navigate to Server Admin > Settings and then search for log. You should see the level to `debug` mode.

Reviewer: ```suggestion
1. To sign in to Grafana, enter `admin` for the username and paste the password that was decoded earlier. 

1. Navigate to **Server Admin > Settings** and then search for `log`. 

   You should see the level to `debug` mode.
```

### Reset Grafana admin secrets (login credentials)

> By default the login credentials for the super admin account are generated via `secrets`.  However, this can be changed easily. To achieve this, use the following steps:

Reviewer: ```suggestion
By default the login credentials for the super admin account are generated via `secrets`. However, you can change this by completing the following steps:
```

1. Edit the `values.yaml` file and search for the string `adminPassword`. There you can define a new password:

   ```yaml
   # Administrator credentials when not using an existing secret (see below)
   adminUser: admin
   adminPassword: admin
   ```

> 1. Then use the `helm upgrade` command as follows:

Reviewer: ```suggestion
1. Run the following `helm upgrade` command:
```

   ```bash
   helm upgrade my-grafana grafana/grafana -f values.yaml -n monitoring
   ```

>    This command will now make your super admin login credentials as `admin` for both username and password.

Reviewer: ```suggestion
   This command makes your super admin login credentials as `admin` for both username and password.
```

1. To verify it, sign in to Grafana, enter `admin` for both username and password. You sould be able to login as super admin.

## Uninstalling the Grafana Deployment

> To uninstall the Grafana deployment, run the command:

Reviewer: ```suggestion
To uninstall the Grafana deployment, run the following command:
```
